advanced-data-types.md

```diff
@@ -0,0 +1,195 @@
+
+```csharp
+// _Name is mapped to 'Name' which is what we use here
+var peters = realm.All<Person>().Filter("Name == 'Peter'");
+```
+
+## Using unpersistable data types
+
+Realm has a wide variety of supported data types - most primitive types in the BCL, as  well as advanced collections, such as sets and dictionaries. But sometimes, you'll come across a data type that Realm can't store yet - the most obvious example being enums. In such cases, you can build on top of the previous technique to expose enum properties in your models and have them be persisted as one of the supported data types:
+
+```csharp
+enum TransactionState
+{
+    Pending,
+    Settled,
+    Error
+}
+
+class Transaction : RealmObject
+{
+    private string _State { get; set; }
+
+    public TransactionState State
+    {
+        get => Enum.Parse<TransactionState>(_State);
+        set => _State = value.ToString();
+    }
+}
+```
+
+Using this technique, you can persist many other types, as long as they can be converted losslessly to a persistable primitive type. In this case, we chose `string`, but we could have just as easily used integer. The string representation takes a bit more memory, but is also more explicit and less error prone - e.g. if you rearrange the enum members, the data will still be consistent.
+
+All that is pretty cool, but we can take it up a notch. By building on top of this idea, we can also devise a strategy for representing complex data types - such as `Vector3` in a Unity game or a `GeoCoordinate` in a location-aware app. To do so, we'll take advantage of embedded objects - a Realm concept that represents a complex data structure that is owned entirely by its parent. Embedded objects are a great fit for this use case because we want to have a strict 1:1 relationship and we want to make sure that deleting the parent also cleans up the embedded objects it owns. Let's see this in action:
+
+```csharp
+class Vector3Model : EmbeddedObject
+{
+    // Casing of the properties here is unusual for C#,
+    // but consistent with the Unity casing.
+    private float x { get; set; }
+    private float y { get; set; }
+    private float z { get; set; }
+
+    public Vector3Model(Vector3 vector)
+    {
+        x = vector.x;
+        y = vector.y;
+        z = vector.z;
+    }
+
+    private Vector3Model()
+    {
+    }
+
+    public Vector3 ToVector3() => new Vector3(x, y, z);
+}
+
+class Powerup : RealmObject
+{
+    [MapTo("Position")]
+    private Vector3Model _Position { get; set; }
+
+    public Vector3 Position
+    {
+        get => _Position?.ToVector3() ?? Vector3.Zero;
+        set => _Position = new Vector3Model(value);
```

Comment:
Positions change a lot (depending on the `GameObject`). If this `PowerUp` is floating around we will create loads of orphaned `EmbeddedObject`s here. Would it be more efficient to update the position instead of creating a new one?
I guess this is - as often - a question of `keeping the tutorial as simple as possible` vs. `our recommended way of implementing things`.

Reply:
I'm not sure if there's a fundamental perf difference between the two options, but I guess people can figure it out themselves - we could just as easily have an `Update(Vector3)` method on the `Vector3Model` where the properties are updated rather than a new object being created. This blog post is intended to be more of an inspiration than an definitive guide for every single use case.